metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java

```diff
@@ -0,0 +1,140 @@
+        final PolarisConnectorDatabaseService polarisConnectorDatabaseService,
+        final HiveConnectorInfoConverter connectorConverter,
+        final ConnectorContext connectorContext
+    ) {
+        this.polarisConnector = polarisConnector;
+        this.polarisConnectorDatabaseService = polarisConnectorDatabaseService;
+        this.connectorConverter = connectorConverter;
+        this.connectorContext = connectorContext;
+        this.catalogName = catalogName;
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public void create(final ConnectorRequestContext requestContext, final TableInfo tableInfo) {
+        final QualifiedName name = tableInfo.getName();
+        try {
+            polarisConnector.createTable(name.getDatabaseName(), name.getTableName());
+        } catch (Exception exception) {
+            // TODO: distinguish between different constraint exceptions once data model supports get APIs
+            throw new ConnectorException(
+                String.format("Failed creating polaris table %s", name), exception);
+        }
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public TableInfo get(final ConnectorRequestContext requestContext, final QualifiedName name) {
+        try {
+            // TODO: implement once data model supports this operation
+            return null;
+        } catch (Exception exception) {
+            throw new ConnectorException(
+                String.format("Failed getting polaris table %s", name), exception);
+        }
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public List<QualifiedName> listNames(
+        final ConnectorRequestContext requestContext,
+        final QualifiedName name,
+        @Nullable final QualifiedName prefix,
+        @Nullable final Sort sort,
+        @Nullable final Pageable pageable
+    ) {
+        try {
+            final List<QualifiedName> qualifiedNames = Lists.newArrayList();
+            final String tableFilter = (prefix != null && prefix.isTableDefinition()) ? prefix.getTableName() : "";
+            for (String tableName : polarisConnector.getTables(name.getDatabaseName(), tableFilter)) {
+                final QualifiedName qualifiedName =
+                    QualifiedName.ofTable(name.getCatalogName(), name.getDatabaseName(), tableName);
+                if (prefix != null && !qualifiedName.toString().startsWith(prefix.toString())) {
+                    continue;
+                }
+                qualifiedNames.add(qualifiedName);
+            }
+            if (sort != null) {
+                ConnectorUtils.sort(qualifiedNames, sort, Comparator.comparing(QualifiedName::toString));
+            }
+            return ConnectorUtils.paginate(qualifiedNames, pageable);
+        } catch (Exception exception) {
+            throw new ConnectorException(
+                String.format("Failed polaris list table names %s using prefix %s", name, prefix), exception);
```

Comment:
we perhaps want to log these exceptions too?

Reply:
Sure, added logging

Reply:
Perhaps this could be logged as well: `"Failed polaris list table names %s using prefix %s"` otherwise just the exception would be too generic to debug maybe.

Reply:
Updated to log that message as well